internal/cmd/controller/controllers/clusterregistration/controller.go

```diff
@@ -180,51 +180,47 @@ func (h *handler) OnChange(request *fleet.ClusterRegistration, status fleet.Clus
 
 	saName := name.SafeConcatName(request.Name, string(request.UID))
 	sa, err := h.serviceAccountCache.Get(cluster.Status.Namespace, saName)
-	if err == nil {
-		if secret, err := h.authorizeCluster(sa, cluster, request); err != nil {
-			return nil, status, fmt.Errorf("failed to authorize cluster: %w", err)
-		} else if secret != nil {
-			status.Granted = true
-			objects = append(objects, secret)
-		}
+	if err != nil {
```

Comment:
should we check if it is a not found error here before creating the sa?

Reply:
Yes